### ISSUE TITLE
[gradle-plugin]: Latest update (5.5.0) is broken on JDK loom eap builds

### ISSUE THREAD

**Issue by sureshg** [diffplug/spotless#693]
- [x] summary of problem
     This [commit](https://github.com/diffplug/spotless/commit/086c84326e3b2e1968ae432e6169e3f10e744618#diff-7a7322bbb00bcfb47ee56dbf3193672bR84) is broken on JDK loom EAP builds (`java.lang.StringIndexOutOfBoundsException`). The plugin was working fine on all JDK eap versions, including project [loom](http://jdk.java.net/loom/) . The version format used is, 

```
$ java -version    
openjdk version "16-loom" 2021-03-16
OpenJDK Runtime Environment (build 16-loom+6-105)
OpenJDK 64-Bit Server VM (build 16-loom+6-105, mixed mode, sharing)
```
Is it possible to handle the EAP version also?

- [x] gradle or maven version
    * `6.7-rc-1`
- [x] spotless version
   * `5.5.0`
- [x] operating system and version
   * Mac OSX (`10.13.6`)
- [x] copy-paste your full Spotless configuration block(s), and a link to a public git repo that reproduces the problem if possible
    
```
spotless {
    java {
        googleJavaFormat(Versions.googleJavaFormat)
    }
  kotlin {
        ktlint(Versions.ktlint).userData(mapOf("disabled_rules" to "no-wildcard-imports"))
        targetExclude("$buildDir/**/*.kt", "bin/**/*.kt")
        // licenseHeader(License.Apache)
    }
  kotlinGradle {
        ktlint(Versions.ktlint).userData(mapOf("disabled_rules" to "no-wildcard-imports"))
        target("*.gradle.kts")
    }
  format("misc") {
        target("**/*.md", "**/.gitignore")
        trimTrailingWhitespace()
        endWithNewline()
    }
 }
 ```
- [x] copy-paste the full content of any console errors emitted by `gradlew spotless[Apply/Check] --stacktrace`
```
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 7
        at com.diffplug.spotless.java.GoogleJavaFormatStep.<clinit>(GoogleJavaFormatStep.java:84)
        ... 199 more
```

**Comment by nedtwigg** [comment]
Ah, this should be easy to fix.

**Comment by nedtwigg** [comment]
Thanks for reporting.  Fixed in `plugin-gradle 5.5.1` and `plugin-maven 2.3.1`
